components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/SQLQueries.java

```diff
@@ -89,6 +89,12 @@ public class SQLQueries {
             "JOIN UM_HYBRID_ROLE m_shared ON s.UM_SHARED_ROLE_ID = m_shared.UM_ID AND s.UM_SHARED_ROLE_TENANT_ID " +
             "= m_shared.UM_TENANT_ID WHERE s.UM_SHARED_ROLE_TENANT_ID = ? AND m_main.UM_UUID IN (";
 
+    public static final String GET_SHARED_ROLE_TO_MAIN_ROLE_MAPPINGS_BY_SUBORG_SQL = "SELECT m_shared.UM_UUID, " +
```

Comment:
You can get this result by querying only UM_SHARED_ROLE table by shared role id column and sub org id colum.
Return the main role id and shared role id columns as the result set

Reply:
- The UUID of a role is stored in the UM_HYBRID_ROLE table.
- The UM_SHARED_ROLE table only contains the UM_IDs for mapping the shared roles.
- Hence, to get the UUIDs of those mapped roles, we have to join the UM_HYBRID_ROLE w.r.t to UM_IDs anyway.
- Therefore, we cannot merely query the UM_SHARED_ROLE table to get the desired output. We have to join the UM_HYBRID_ROLE table as well.